### PR TITLE
Transition world_location_news to design system

### PR DESCRIPTION
**Pull request by chriswhitehouse-tw**
## Description

This ports the world_location_news index page to the GOV.UK Design System. It appends legacy to the current view. Then adds the new view in the GOV.UK Design System and conditionally renders the Bootstrap or GOV.UK Design System based on whether the user has the "Preview design system" permission.

## Screenshots
Screenshot 2023-03-17 at 14 55 27
Screenshot 2023-03-17 at 14 55 35

Screenshot 2023-03-21 at 10 37 19

Screenshot 2023-03-21 at 10 37 23

## Trello

https://trello.com/c/DdEO3vou

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: app/views/admin/world_location_news/_world_location_table.html.erb [hunk 1]
```diff
@@ -1,47 +1,52 @@
...
+  ],
+  rows:
+    world_locations.map do |world_location|
+      [
+        {
+          text: world_location.display_type
+        },
+        {
+          text:
+            link_to(world_location.name, [:admin, world_location.world_location_news], class: "govuk-link") << "<br><br>".html_safe <<
```
Comment (davidgisbey) [inline app/views/admin/world_location_news/_world_location_table.html.erb:28]: This definitely works, but maybe something like this is a little easier to parse 

```
tag.p(link_to(world_location.name, [:admin, world_location.world_location_news], class: "govuk-link"), class: "govuk-body govuk-!-margin-bottom-4") +
              if world_location.world_location_news.mission_statement.present?
                format_with_html_line_breaks(world_location.world_location_news.mission_statement)
              else
                sanitize("<span><em>No mission statement yet, </em>#{link_to("create one", edit_admin_world_location_news_path(world_location.world_location_news), class: "govuk-link")}.")
              end
```

Neither of the cases you called html_safe rang any alarm bells in terms of introducing xss vulnerabilities though. Generally i use `sanitize` where possible instead though https://apidock.com/rails/ActionView/Helpers/SanitizeHelper/sanitize

**Comment by davidgisbey** [comment]
Sorry Chris i just noticed one more thing. You've ported the `_world_location_table.html.erb` partial to use the GOV.UK Design System implementation, but it wasn't duplicated and one renamed to legacy for the Bootstrap implementation.

This means it'll get rendered for both implementations. When i remove the "Preview design system" permission this is what i see for the Bootstrap implementation.

image

To fix this we can do the following. Duplicate the partial and prepend one with legacy. Retain the previous implementation in the legacy one. Then call the legacy partial in the legacy index view

```
<%= render "legacy_world_location_table",  title: "Active", world_locations: @active_world_locations %>

<%= render "legacy_world_location_table",  title: "Inactive", world_locations: @inactive_world_locations %>

```

**Comment by cbernardojr-gds** [comment]
@chriswhitehouse-tw As discussed:

World location table
- Option 1: Separate Location and Details in the table in 2 column
- Option 2: Remove the location link and make it bold (keep the details in the same column)
- Create a column on the right with an Edit link for the Edit world location Details page/tab
- Keep the link on translations to Edit translation

Active and Inactive tables:
- Option 1: Use the filter radios. Reference: https://gds-wh.herokuapp.com/v2/publication/history?pageState=empty&documentState=draft&pageMessage=&prototypeVersion=v2 Prototype password: wh1t3h@11
- Option 2: Use tabs. Reference: https://gds-wh.herokuapp.com/v2/organisation/features#documents Prototype password: wh1t3h@11

File: app/views/admin/world_location_news/_world_location_table.html.erb (hunk 1 above)
Comment (chriswhitehouse-tw) [reply app/views/admin/world_location_news/_world_location_table.html.erb]: Nice, thanks David. 👍 Much better code. I've implemented as suggested.

File: app/views/admin/world_location_news/_world_location_table.html.erb [hunk 2]
```diff
@@ -1,47 +1,52 @@
...
+              else
+                "<span><em>No mission statement yet, </em>#{link_to("create one", edit_admin_world_location_news_path(world_location.world_location_news), class: "govuk-link")}."
+              end.html_safe
+        },
+        {
+          text: world_location.active? ? "active" : "inactive"
+        },
+        {
+          text:
+            if world_location.non_english_translated_locales.any?
```
[the remark this answers is not in the document]
Comment (chriswhitehouse-tw) [reply app/views/admin/world_location_news/_world_location_table.html.erb]: Ditto, much better. And implemented.

**Comment by davidgisbey** [comment]
Excellent work, just left a couple of minor comments.

File: app/views/admin/world_location_news/_world_location_table.html.erb [hunk 3]
```diff
@@ -44,8 +42,10 @@
             else
               "none"
             end)
+        },
+        {
+          text: link_to("Edit", [:admin, world_location.world_location_news], class: "govuk-link")
```
[the remark this answers is not in the document]
Comment (chriswhitehouse-tw) [reply app/views/admin/world_location_news/_world_location_table.html.erb]: Nice!

**Comment by DilwoarH** [comment]
Screenshot 2023-03-20 at 11 12 07 am

Should we have a "No active news"?

**Review by DilwoarH — APPROVED** [review]

**Comment by DilwoarH** [comment]
Great work @chriswhitehouse-tw ! 🥇

**Pull request merged**